### PR TITLE
[Cirrus Polls] Add PollsCollection

### PR DESCRIPTION
**Pull request by quantumagi**
See https://app.clickup.com/t/40ra67.

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.PoA/Voting/VotingData.cs [hunk 1]
```diff
@@ -76,7 +76,14 @@ public override bool Equals(object obj)
         /// <inheritdoc />
         public override int GetHashCode()
         {
-            return this.Data.GetHashCode() ^ this.key;
+            ulong hash = 17;
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.Features.PoA/Voting/VotingData.cs:79]: Why this change @quantumagi ?

Comment (quantumagi) [reply src/Stratis.Bitcoin.Features.PoA/Voting/VotingData.cs:79]: As far as I can remember the `GetHashCode` is based on object reference while we're instead interested in the content matching. Will refresh my memory on this.

Comment (fassadlr) [reply src/Stratis.Bitcoin.Features.PoA/Voting/VotingData.cs:79]: This wont have any effect on existing saved polls or anything like that? I suppose it shouldnt but just double checking.

**Review by fassadlr — APPROVED** [review]
